modelgauge/annotation_pipeline.py

```diff
@@ -0,0 +1,163 @@
+import csv
+import jsonlines
+import sys
+import traceback
+from abc import abstractmethod, ABCMeta
+from collections import defaultdict
+from dataclasses import dataclass
+from pydantic import BaseModel
+from typing import Iterable
+
+from modelgauge.annotator import Annotator
+from modelgauge.pipeline import Source, Pipe, Sink
+from modelgauge.prompt import TextPrompt
+from modelgauge.prompt_pipeline import PromptOutput
+from modelgauge.single_turn_prompt_response import PromptWithContext
+from modelgauge.sut import PromptResponseSUT, SUTCompletion
+
+
+@dataclass
+class AnnotatorInputSample:
+    prompt: PromptWithContext
+    sut_uid: str
+    response: SUTCompletion
+
+    def __hash__(self):
+        return hash(self.prompt.source_id + self.sut_uid)
+
+
+class AnnotatorInput(metaclass=ABCMeta):
+    @abstractmethod
+    def __iter__(self) -> Iterable[AnnotatorInputSample]:
+        pass
+
+    def __len__(self):
+        count = 0
+        for prompt in self:
+            count += 1
+        return count
+
+
+class CsvAnnotatorInput(AnnotatorInput):
+    def __init__(self, path):
+        super().__init__()
+        self.path = path
+
+    def __iter__(self) -> Iterable[AnnotatorInputSample]:
+        with open(self.path, newline="") as f:
+            csvreader = csv.DictReader(f)
+            for row in csvreader:
+                prompt = PromptWithContext(
```

Comment:
prompt, response as separate columns fine for now. multiturn out of scope for now, but this may need to change in the futreu